stdlib/public/Python/Python.swift

```diff
@@ -1257,6 +1257,30 @@ extension PythonObject : MutableCollection {
   }
 }
 
+extension PythonObject : Sequence {
+  public struct Iterator : IteratorProtocol {
+    fileprivate let pythonIterator: PythonObject
+
+    public func next() -> PythonObject? {
+      guard let result = PyIter_Next(self.pythonIterator.borrowedPyObject) else {
+        try! throwPythonErrorIfPresent()
+        return nil
+      }
+
+      defer {
+        Py_DecRef(result)
+      }
+
+      return PythonObject(consuming: result)
+    }
+  }
+
+  public func makeIterator() -> Iterator {
+    let result = PyObject_GetIter(self.borrowedPyObject)
```

Comment:
If `PyObject_GetIter` returns `nil` (aka, a `NULL` pointer), this would be a precondition failure and a Python `TypeError` must have been raised, according to [`PyObject_GetIter`'s documentation](https://docs.python.org/3/c-api/object.html#c.PyObject_GetIter)). Instead of force-unwrapping `result`, we should use a guard statement and throw the Python exception.

```swift
guard let result = PyObject_GetIter(borrowedPyObject) else {
  try! throwPythonErrorIfPresent()
  // Unreachable. A Python `TypeError` must have been thrown.
  preconditionFailure()
}
return Iterator(pythonIterator: PythonObject(consuming: result))
```